### PR TITLE
adding extra Archive Tests, adding contract annotations for ArchiveLayout

### PR DESCRIPTION
**Pull request by slorello89**
Adding some missing test cases for the ArchiveLayout StartArchive paths.

### CODE REVIEW

**Comment by slorello89** [comment]
#148 Pointed out to me a couple of missing test paths for the SDK here. It doesn't appear that the code path they suggested is failing, still waiting on clarity - but it's still worth adding the cases.

**Comment by slorello89** [comment]
After further discussion with the originator of #148 it's clear what's going on here - when the stylesheet is being set to an empty string it's erroneously being serialized with the rest of the `ArchiveLayout` class and shipped off to the OpenTok API, which rejects the request with a 400 error. I've added some new annotations to the Archive layout (as well as new tests to OpenTokTest) to make sure this is being done correctly.

File: OpenTokTest/OpenTokTest.cs [hunk 1]
```diff
@@ -649,7 +649,7 @@ public void TestArchiveNonCustomLayout()
             var expectedString = @"{""sessionId"":""abcd12345"",""name"":""an_archive_name"",""hasVideo"":true,""hasAudio"":true,""outputMode"":""composed"",""layout"":{""type"":""pip""}}";
             var httpClient = new HttpClient();
             var data = new Dictionary<string, object>() { { "sessionId", "abcd12345" }, { "name", "an_archive_name" }, { "hasVideo", true }, { "hasAudio", true }, { "outputMode", "composed" } };
-            var layout = new ArchiveLayout { Type = LayoutType.pip };
+            var layout = new ArchiveLayout { Type = LayoutType.pip};
```
Comment (dragonmantank) [inline OpenTokTest/OpenTokTest.cs:652]: This is unbalanced now and sticks out :P

**Review by dragonmantank — DISMISSED** [review]
LGTM :+1:

**Review by dragonmantank — APPROVED** [review]
Re-adding approval since it was cleared after the update from master

**Pull request merged**